packages/dara-core/dara/core/interactivity/derived_data_variable.py

```diff
@@ -297,6 +298,11 @@ async def get_data(
 
         # Run the filtering
         data = await cls._filter_data(data, count_cache_key, data_entry, store, filters, pagination)
+        if format_for_display and data is not None:
+            for col in data.columns:
+                if data[col].dtype == 'object':
+                    # We need to convert all values to string to avoid issues with displaying data in the Table component, for example when displaying datetime and number objects in the same column
+                    data[col] = data[col].apply(str)
```

Comment:
One thing I just thought about - is this operating on a copy of the data or the original one? Could you double check whether that's the case, e.g. if you have a Table displaying a DataVariable/DerivedDataVariable and also a different e.g. DerivedVariable consuming the data, is it not affected?